product_template_pricelist_button/models/product_template.py

```diff
@@ -0,0 +1,31 @@
+# © 2024 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3.0 (https://www.gnu.org/licenses/lgpl-3.0.html)
+
+from odoo import models
+
+class ProductTemplate(models.Model):
+    _inherit = 'product.template'
+
+    def open_pricelist_rules(self):
+        action = super().open_pricelist_rules()
+        self.ensure_one()
+        domain = ['|','|',
+            ('product_tmpl_id', '=', self.id),
+            ('product_id', 'in', self.product_variant_ids.ids),
+            ('categ_id','<=',self.categ_id.id),
+            ]
+        action["domain"] = domain
+        return action
+
+    def _compute_item_count(self):
+        super()._compute_item_count()
+        for template in self:
+            template.pricelist_item_count = template.env['product.pricelist.item'].search_count([
```

Comment:
Not sure if this domain (despite of wrong operator) should be the right one, this is always difficult to solve. Please first check with @ChristianSantamaria , if possible

Reply:
It should be that domain, because when the button is defined in the view with the open_pricelist_rules action, it shows the pricelist_item_count field, which is the computed field of the _compute_item_count function.
![imagen](https://github.com/user-attachments/assets/2c1f3de3-c580-4628-a049-7ca347f6ab02)
![imagen](https://github.com/user-attachments/assets/fe0a9f73-e635-4267-beb5-82ad0c398f6b)